lua/nvim-tree/explorer/utils.lua

```diff
@@ -17,7 +17,16 @@ function M.node_comparator(a, b)
     return false
   end
 
-  return a.name:lower() <= b.name:lower()
+  if M.config.sort_by == "modification_time" then
+      -- newer files have higher mtime
+      local a_mtime = uv.fs_stat(a.absolute_path, nil).mtime.sec
+      local b_mtime = uv.fs_stat(b.absolute_path, nil).mtime.sec
+      return b_mtime <= a_mtime
+  elseif M.config.sort_by == "name" then
+    return a.name:lower() <= b.name:lower()
+  else
+    return a.name:lower() <= b.name:lower()
+  end
```

Comment:
can you maybe just generate a different node_comparator functions based on the config instead of doing checks inside this ?

Reply:
Sure. Any changes requested for the config names?

Reply:
Config name is good enough :)